### ISSUE TITLE
attr-name-style isn't template friendly

### ISSUE THREAD

**Issue by coalman** [htmllint/htmllint#98]
The current options for `attr-name-style` don't work well for angular and thymeleaf templates where characters like `:` and `_` are allowed.

Perhaps another choice that allows these characters, or another option to support template languages that are syntactically similar to html.

**Comment by waffledonkey** [comment]
I'm also struggling with this.  "lowercase" seems to fail on things like aria-label, etc. which is to be expected, but when you switch to "dash" the error is: 

`Fatal error: op is not defined` which is an altogether different error than if you throw in a random format string.  In other words it seems like a bug more than a doc problem.  

It seems like either allowing for error levels (non-build-breaking warnings for attr-name-style) or regex or an array of allowed keyword styles or something.  As it is I had no option but to switch to false.

**Comment by opus1269** [comment]
Google's Polymer Web Components is another example. It uses the following syntax to bind to attributes:

```html
<div class="setting-label" hidden$="[[!mainLabel]]">{{mainLabel}}</div>
```

**Comment by mlochbaum** [comment]
`attr-name-ignore-regex` mostly solves this issue. Between that and the fact that `attr-name-style` itself can now accept a RegExp, I would say this is solved.
